pkg/data-source-validator/data_source_validator.go

```diff
@@ -93,9 +94,16 @@ func NewDataSourceValidator(
 
 	pvcInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc:    func(obj interface{}) { ctrl.enqueueWork(obj) },
-			UpdateFunc: func(oldObj, newObj interface{}) { ctrl.enqueueWork(newObj) },
-			DeleteFunc: func(obj interface{}) { ctrl.enqueueWork(obj) },
+			AddFunc: ctrl.enqueueWork,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				newPvc := newObj.(*v1.PersistentVolumeClaim)
+				oldPvc := oldObj.(*v1.PersistentVolumeClaim)
+				if newPvc.ResourceVersion == oldPvc.ResourceVersion {
+					return
```

Comment:
Do you see this happening without this check?  I meant it will add to the queue even if resource version is the same?

Reply:
Oh! That's news to me. I didn't know it could be disabled entirely.

I feel safe with this optimization, and I'm not sure if disabling resyncs adds any other corner cases I'm not aware of.

Reply:
The resync period can be set to zero or the simpler AddEventHandler can be used instead of AddEventHandlerWithResyncPeriod: https://pkg.go.dev/k8s.io/client-go/tools/cache#SharedInformer

As you filter out exactly the work that gets added by AddEventHandlerWithResyncPeriod, I think the better solution is to use AddEventHandler and avoid the overhead of iterating over all objects.

Reply:
Okay I'll trust you're right.

Reply:
@bswartz do you still need to do this check after switching to AddEventHandler()?

Reply:
> do you still need to do this check after switching to AddEventHandler()?

Yes I tested it that way and I was still getting duplicate updates. They were less frequent, but not zero. The current commit completely prevents duplicate updates.